src/app.py

```diff
@@ -78,90 +81,312 @@ def index():
+be valid JSON itself. No key/value pair is stored unless all key/value pairs are stored, as indicated
+by an HTTP 200 Response.
+
+Parameters
+----------
+None
+
+Returns
+-------
+HTTP 200 Response with Content-Type application/json
+The JSON body is a dictionary with an entry confirming all the key/value pairs were stored.
+
+HTTP 400 Response with Content-Type application/json
+The JSON body is a dictionary with an error message describing the problem that kept
+each key/value pair from being stored.  Examples causes are
+- A body in the Request which is not valid JSON for a dictionary of key/value pairs
+- Failure to specify a Content-Type header of application/json for the Request
+- A body in the Request which is valid JSON but is empty
+- Keys which are too long
+- Keys which contain whitespace or other disallowed characters
+Key/value pairs in the Request which are absent from error message can be assumed to
+pass validations, but were not stored if the HTTP Response is not 200.
+
+HTTP 500 Response
+An unexpected error in the server, including unexpected problems with the data store.
+"""
+@app.route(rule='/user/keys', methods=["PUT"])
+@secured(has_write=True)
+def upsert_key_values():
+    global ukv_worker
+
+    try:
+        success_msg = ukv_worker.upsert_key_values(req=request)
+        return jsonify({'message': success_msg})
+    except (ukvEx.UKVValueFormatException, ukvEx.UKVRequestFormatException) as e_400:
+        return jsonify({'error': e_400.message}), 400
+    except (ukvEx.UKVBadKeyListException) as e_400:
+        return jsonify(e_400.data), 400
+    except (ukvEx.UKVDataStoreQueryException, Exception) as e_500:
+        logger.exception(f"Unexpected error setting key/value pair(s) in JSON payload={request.get_json()}")
+        return jsonify({'error': f"Unexpected error setting key/value pair(s). See logs."}), 500
+
+"""
+An endpoint to delete a key/value pair for the authenticated user.
+On success returns an HTTP 200 Response.
+
+Parameters
+----------
+key : The key delete for the authenticated user, as specified on the API endpoint 
+
+Returns
+-------
+HTTP 200 Response with Content-Type application/json
+The JSON body is a dictionary with an entry confirming the key/value pair was deleted.
+
+HTTP 400 Response with Content-Type application/json
+The JSON body is a dictionary with an error message describing the problem that kept
+each key/value pair from being deleted.  Examples causes are
+- Keys which are too long
+- Keys which contain whitespace or other disallowed characters
+
+HTTP 404 Response with Content-Type application/json
+The JSON body is a dictionary with an error message indicating the key was not found in the data store.
+
+HTTP 500 Response
+An unexpected error in the server, including unexpected problems with the data store.
+"""
+@app.route(rule='/user/keys/<key>', methods=["DELETE"])
+@secured(has_write=True)
+def delete_key_value(key: Annotated[str, 50]):
```

Comment:
To avoid confusion, does `Annotated[str, 50]` suggest the max length is 50 or the key length needs to be exact 50?

Reply:
It is just type hinting introduced in Python 3.10 for readability, documentation, and static code analysis.  Not "enforced", so the explicit verification of length remains in the code.  When we were at Python 3.9, I would have just used `str` for the hint.